Add documentation on maven.config comments

ctubbsii opened the pull request:
Update the configure page to explain how to write comments in the maven.config file.

Following this checklist to help us incorporate your
contribution quickly and easily:

- [x] Your pull request should address just one issue, without pulling in other changes.
- [x] Each commit in the pull request should have a meaningful subject line and body.
  Note that commits might be squashed by a maintainer on merge.
- [x] Run `mvn site` and examine output in `target/site` directory.
  Site will also be built on your pull request automatically and attached to GitHub Action result.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

- [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
- [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Bukama commented on content/markdown/configure.md:
```diff
@@ -85,10 +85,13 @@ For example things like `-T3 -U --fail-at-end`. So you only have to call Maven b
 clean package` instead of `mvn -T3 -U --fail-at-end clean package` and not to miss the `-T3 -U --fail-at-end` options on every call.
 The `.mvn/maven.config` is located in the project's top level `.mvn` directory and also works in the root of a multi module build.
 
+Comments can be embedded by starting a line with `#`.
```
> Since Maven 3.9.x comments can be embedded by starting a line with `#`. Maven 3.8.x and older versions do not support comments in this file.

You speak of multiple versions here so `does` => `do`.

In general I like the suggestion by @cstamas to make it clear that older versions (which are still used a lot out there as the audiances of my talks state :( ) do not support it.

ctubbsii replied:
Done in ebefb6cca073da922bbe0b8fea9b125e43128bbc

ctubbsii commented:
I've rebased and updated the PR to reflect the code review suggestions. Instead of having a separate notice line, I incorporated the explanation into the existing NOTICE line.
